velero-plugins/configmap/restore.go

```diff
@@ -0,0 +1,39 @@
+package configmap
+
+import (
+	"github.com/konveyor/openshift-velero-plugin/velero-plugins/common"
+	"github.com/sirupsen/logrus"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+// RestorePlugin is a restore item action plugin for Velero
+type RestorePlugin struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo returns a velero.ResourceSelector that applies to configmaps
+func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"configmaps"},
+	}, nil
+}
+
+// Execute action for the restore plugin for the configmap resource
+// We want to exclude from backup configmaps that contain ownerReference to build pods
+func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	p.Log.Info("[cm-restore] Entering ConfigMap restore plugin")
+	meta, err := meta.Accessor(input.Item)
+	if err != nil {
+		p.Log.Warnf("[cm-restore] Unable to accessed metadata, err: %v", err)
+	}
+	annotations := meta.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
```

Comment:
Since we're not setting any new annotations here, I don't think we need this step. We just need to add a nil check to L34.

Reply:
Or just a return statement here -- if there are no annotations, then we know we don't have the skip annotation.

Reply:
get rid of the make step? keep the nil check?

Reply:
Yes -- if it's nil, then we know the annotation isn't there, so we can just exit now and not skip.